File: src/services/StorageService.ts
import AsyncStorage from '@react-native-async-storage/async-storage';

export interface TimesheetEntry {
  id: number;
  date: string;
  timeIn: string;
  timeOut: string;
  breakMinutes: string;
  task: string;
  day: string;
  hours: string;
}

export interface AppData {
  employeeName: string;
  studentNumber: string;
  entries: TimesheetEntry[];
  lastSaved: string;
}

const STORAGE_KEY = 'timesheet_app_data';

export const StorageService = {
  async saveData(data: Partial<AppData>): Promise<void> {
    try {
      const existingData = await this.loadData();
      const updatedData = {
        ...existingData,
        ...data,
        lastSaved: new Date().toISOString(),
      };
      await AsyncStorage.setItem(STORAGE_KEY, JSON.stringify(updatedData));
    } catch (error) {
      console.error('Error saving data:', error);
      throw error;
    }
  },

  async loadData(): Promise<AppData> {
    try {
      const data = await AsyncStorage.getItem(STORAGE_KEY);
      if (data) {
        return JSON.parse(data);
      }
      return {
        employeeName: '',
        studentNumber: '',
        entries: [],
        lastSaved: '',
      };
    } catch (error) {
      console.error('Error loading data:', error);
      return {
        employeeName: '',
        studentNumber: '',
        entries: [],
        lastSaved: '',
      };
    }
  },

  async clearAllData(): Promise<void> {
    try {
      await AsyncStorage.removeItem(STORAGE_KEY);
    } catch (error) {
      console.error('Error clearing data:', error);
      throw error;
    }
  },
};

File: mitl_timesheet_flutter/lib/services/storage_service.dart
import 'dart:convert';
import 'package:shared_preferences/shared_preferences.dart';
import '../models/app_data.dart';
import '../models/timesheet_entry.dart';

class StorageService {
  static const String _storageKey = 'timesheet_app_data';

  static Future<void> saveData(AppData data) async {
    try {
      final prefs = await SharedPreferences.getInstance();
      final jsonString = jsonEncode(data.toJson());
      await prefs.setString(_storageKey, jsonString);
    } catch (error) {
      print('Error saving data: $error');
      throw error;
    }
  }

  static Future<AppData> loadData() async {
    try {
      final prefs = await SharedPreferences.getInstance();
      final jsonString = prefs.getString(_storageKey);
      
      if (jsonString != null) {
        final jsonData = jsonDecode(jsonString);
        return AppData.fromJson(jsonData);
      }
      
      return AppData(
        employeeName: '',
        studentNumber: '',
        entries: [],
        lastSaved: '',
      );
    } catch (error) {
      print('Error loading data: $error');
      return AppData(
        employeeName: '',
        studentNumber: '',
        entries: [],
        lastSaved: '',
      );
    }
  }

  static Future<void> clearAllData() async {
    try {
      final prefs = await SharedPreferences.getInstance();
      await prefs.remove(_storageKey);
    } catch (error) {
      print('Error clearing data: $error');
      throw error;
    }
  }
}
